namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java

```diff
@@ -422,9 +424,20 @@ public void scanNotActiveBroker() {
             long last = next.getValue().getLastUpdateTimestamp();
             if ((last + BROKER_CHANNEL_EXPIRED_TIME) < System.currentTimeMillis()) {
                 RemotingUtil.closeChannel(next.getValue().getChannel());
-                it.remove();
-                log.warn("The broker channel expired, {} {}ms", next.getKey(), BROKER_CHANNEL_EXPIRED_TIME);
-                this.onChannelDestroy(next.getKey(), next.getValue().getChannel());
+                try {
+                    try {
+                        this.lock.writeLock().lockInterruptibly();
+                        it.remove();
+                        log.warn("The broker channel expired, {} {}ms", next.getKey(), BROKER_CHANNEL_EXPIRED_TIME);
+                        this.onChannelDestroy(next.getKey(), next.getValue().getChannel());
+                    } catch (Exception e) {
+                        log.error("Delete Not Active Broker Exception", e);
+                    } finally {
+                        this.lock.writeLock().unlock();
+                    }
+                } catch (Exception e) {
+                    log.error("Delete Not Active Broker Exception", e);
+                }
```

Comment:
Nesting of try-catch still exists. 
IMO, maybe moving the writelock out of the loop is a good choice.